tests/vxlan/test_vxlan.py

```diff
@@ -0,0 +1,645 @@
+
+'''
+    Script to automate the cases listed in VxLAN HLD document:
+    https://github.com/Azure/SONiC/blob/8ca1ac93c8912fda7b09de9bfd51498e5038c292/doc/vxlan/Overlay%20ECMP%20with%20BFD.md#test-cases
+
+    To test functionality:
+    ./run_tests.sh -n ucs-m5-2 -d mth64-m5-2 -O -u -e -s -e --disable_loganalyzer -m individual -p /home/vxr/vxlan/logs/    -c 'vxlan/test_vxlan.py'
+
+    To test ECMP with 2 paths per destination:
+    ./run_tests.sh -n ucs-m5-2 -d mth64-m5-2 -O -u -e -s -e --disable_loganalyzer -m individual -p /home/vxr/vxlan/logs/    -c 'vxlan/test_vxlan.py' -e '--nhs_per_destination=2'
+
+    To test ECMP+Scale:
+    ./run_tests.sh -n ucs-m5-2 -d mth64-m5-2 -O -u -e -s -e --disable_loganalyzer -m individual -p /home/vxr/vxlan/logs/    -c 'vxlan/test_vxlan.py' \
+                    -e '--ecmp_nhs_per_destination=128' -e '--total_number_of_nexthops=128000'
+
+    Other options:
+        ecmp_nhs_per_destination    : Number of ECMP next-hops per destination.
+        total_number_of_endpoints : Number of Endpoints (a pool of this number of ip addresses will used for next-hops).
+        total_number_of_nexthops    : Maximum number of all nexthops for every destination combined(per encap_type).
+        vxlan_port                                : Global vxlan port (UDP port) to be used for the DUT. Default: 4789
+'''
+
+import time
+import re
+import ipaddress
+import json
+import logging
+import random
+from datetime import datetime
+from sys import getsizeof
+
+import pytest
+
+from tests.common.fixtures.ptfhost_utils import copy_ptftests_directory  # lgtm[py/unused-import]
+from tests.ptf_runner import ptf_runner
+
+Logger = logging.getLogger(__name__)
+
+# Some of the Constants used in this script.
+Constants = {}
+# Should I keep the temporary files copied to DUT or PTF ?
+Constants['KEEP_TEMP_FILES'] = False
+# Is debugging going on, or is it a production run? If it is a
+# production run, use time-stamped file names for temp files.
+Constants['DEBUG'] = False
+# The host id in the ip addresses for DUT. It can be anything,
+# but helps to keep as a single number that is easy to identify
+# as DUT.
+Constants['DUT_HOSTID'] = 1
+
+# Mapping the version to the python module.
+IP_TYPE = {
+    'v4' : ipaddress.IPv4Address,
+    'v6' : ipaddress.IPv6Address
+}
+# This is the mask values to use for destination
+# in the vnet routes.
+HOST_MASK = {'v4' : 32, 'v6' : 128}
+
+# This is the mask to use in the DUT ip address
+NET_MASK = {'v4' : 24, 'v6' : 100}
+
+# This is the list of encapsulations that will be tested in this script.
+# v6_in_v4 means: V6 payload is encapsulated inside v4 outer layer.
+# This list is used in many locations in the script.
+SUPPORTED_ENCAP_TYPES = ['v4_in_v4', 'v4_in_v6', 'v6_in_v4', 'v6_in_v6']
+
+pytestmark = [
+    # This script supports any T1 topology: t1, t1-64-lag, t1-lag.
+    pytest.mark.topology("t1"),
+    pytest.mark.sanity_check(post_check=False),
+    pytest.mark.asic("cisco-8000")
```

Comment:
The next iteration of this script will turn on the post_check as well. It is good to keep the sanity_check, and also it is present in the previous test_vnet_vxlan.py as well.
As of now we support this script for Cisco platform only, so the ASIC check is needed.

Reply:
I still do not think this test should be supported only by Cisco platforms. Even it might not pass on other platforms at this time, I believe it should be made general and maybe mark other platforms as expected failure (if not supported) later.

Reply:
I will remove the cisco check.